griptape/drivers/prompt/openai_chat_prompt_driver.py

```diff
@@ -131,47 +139,152 @@ def _base_params(self, prompt_stack: PromptStack) -> dict:
             # JSON mode still requires a system message instructing the LLM to output JSON.
             prompt_stack.add_system_message("Provide your response as a valid JSON object.")
 
-        messages = self._prompt_stack_to_messages(prompt_stack)
+        messages = self.__to_openai_messages(prompt_stack.messages)
 
         params["messages"] = messages
 
         return params
 
-    def __to_role(self, message: Message) -> str:
+    def __to_openai_messages(self, messages: list[Message]) -> list[dict]:
+        openai_messages = []
+
+        for message in messages:
+            if message.is_text():
+                openai_messages.append({"role": message.role, "content": message.to_text()})
+            elif message.has_any_content_type(ActionResultMessageContent):
```

Comment:
Can a message ever have a mix of `ActionResultMessageContent` and other message content types?

Reply:
As of right now, OpenAi exclusively returns function results -- so no there will not be a mix.

I'm not sure I follow your second question though.

Reply:
That's fine, this was not a very important comment/question